pkg/segment/aggregations/segaggs.go

```diff
@@ -2408,6 +2414,665 @@ func performValueColRequest(nodeResult *structs.NodeResult, aggs *structs.QueryA
+		return float64(num), true
+	case int8:
+		return float64(num), true
+	case int16:
+		return float64(num), true
+	case int32:
+		return float64(num), true
+	case int64:
+		return float64(num), true
+	case uint:
+		return float64(num), true
+	case uint8:
+		return float64(num), true
+	case uint16:
+		return float64(num), true
+	case uint32:
+		return float64(num), true
+	case uint64:
+		return float64(num), true
+	case float32:
+		return float64(num), true
+	case float64:
+		return num, true
+	case string:
+		floatVal, err := strconv.ParseFloat(num, 64)
+		if err != nil {
+			return 0.0, false
+		}
+		return floatVal, true
+	default:
+		return 0.0, false
+	}
+}
+
+// Get the float/numeric value from the record or fieldToValue map if possible
+// Should pass either record or fieldToValue
+func getFloatValForBin(fieldToValue map[string]segutils.CValueEnclosure, record map[string]interface{}, field string) (float64, error) {
+	var err error
+	var fieldValue interface{}
+	var exist bool
+	if record != nil {
+		fieldValue, exist = record[field]
+		if !exist {
+			return 0, fmt.Errorf("getFloatValForBin: field %s does not exist in record", field)
+		}
+	} else {
+		fieldCValue, exist := fieldToValue[field]
+		if !exist {
+			return 0, fmt.Errorf("getFloatValForBin: field %s does not exist in record", field)
+		}
+		fieldValue = fieldCValue.CVal
+	}
+
+	fieldValueFloat, isNumeric := getNumericValue(fieldValue)
+	if !isNumeric {
+		return 0, fmt.Errorf("getFloatValForBin: field %s is not a numeric, err: %v", field, err)
+	}
+
+	return fieldValueFloat, nil
+}
+
+// Function to find the span range length
+func findSpan(minValue float64, maxValue float64, maxBins uint64, minSpan *structs.BinSpanLength, field string) *structs.BinSpanOptions {
+	if field == "timestamp" {
+		return findEstimatedTimeSpan(minValue, maxValue, maxBins, minSpan)
+	}
+	if minValue == maxValue {
+		return &structs.BinSpanOptions{
+			BinSpanLength: &structs.BinSpanLength{
+				Num:       1,
+				TimeScale: segutils.TMInvalid,
+			},
+		}
+	}
+
+	// span ranges estimated are in powers of 10
+	span := (maxValue - minValue) / float64(maxBins)
+	exponent := math.Log10(span)
+	exponent = math.Ceil(exponent)
+	spanRange := math.Pow(10, exponent)
+
+	// verify if estimated span gives correct number of bins, refer the edge case like 301-500 for bins = 2
+	for {
+		lowerBound, _ := getBinRange(minValue, spanRange)
+		_, upperBound := getBinRange(maxValue, spanRange)
+
+		if (upperBound-lowerBound)/spanRange > float64(maxBins) && spanRange <= math.MaxFloat64/10 {
+			spanRange = spanRange * 10
+		} else {
+			break
+		}
+	}
+
+	// increase the spanRange till minSpan is satisfied
+	if minSpan != nil {
+		for {
+			if spanRange < minSpan.Num && spanRange <= math.MaxFloat64/10 {
+				spanRange = spanRange * 10
+			} else {
+				break
+			}
+		}
+	}
+
+	return &structs.BinSpanOptions{
+		BinSpanLength: &structs.BinSpanLength{
+			Num:       spanRange,
+			TimeScale: segutils.TMInvalid,
+		},
+	}
```

Comment:
Yeah it is correct to use Invalid because we are operating on numerical data and not timedata. If the field is timestamp we would get the estimated timespan from findEstimatedTimeSpan method in line 2484.